sympy/concrete/products.py

```diff
@@ -399,6 +399,21 @@ def _eval_product_direct(self, term, limits):
         (k, a, n) = limits
         return Mul(*[term.subs(k, a + i) for i in range(n - a + 1)])
 
+    def _eval_derivative(self, x):
+        from sympy.concrete.summations import Sum, Dummy, Symbol, Derivative
+        if isinstance(x, Symbol) and x not in self.free_symbols:
+            return S.Zero
+        f, limits = self.function, list(self.limits)
+        limit = limits.pop(-1)
+        if limits:
+            f = self.func(f, *limits)
+        _, a, b = limit
```

Comment:
If we are using the variable called `_`, then it is probably better to give it a different name. Maybe `i` or `x`.

Reply:
Other than these two small things, I think it's good to go.

Reply:
> If we are using the variable called `_`, then it is probably better to give it a different name. Maybe `i` or `x`.
Sure I replaced it by `i`.
Thanks.